Docs/source/building/openpmd.rst

```diff
@@ -16,74 +16,90 @@ library from source.
 
 In order to install spack, you can simply do:
 
-::
+.. code-block:: bash
 
-  git clone https://github.com/spack/spack.git
-  export SPACK_ROOT=/path/to/spack
-  . $SPACK_ROOT/share/spack/setup-env.sh
+   git clone https://github.com/spack/spack.git
+   export SPACK_ROOT=/path/to/spack  # note: adjust this path
+   . $SPACK_ROOT/share/spack/setup-env.sh
 
 (You may want to add the last 2 lines to your ``.bashrc`` file.)
 
 
-Building openPMD support, by installing openPMD-api directly from spack
------------------------------------------------------------------------
+WarpX Development Environment with Spack
+----------------------------------------
 
-First, install the openPMD-api library:
+Create an activate a Spack environment with all software needed to build WarpX
 
-::
+.. code-block:: bash
 
-    spack install openpmd-api -python
+   spack env create warpx-dev    # you do this once
+   spack env activate warpx-dev
+   spack add gmake
+   spack add mpi
+   spack add openpmd-api
```

Comment:
The change to the openpmd-api line is just because the `package.py` does not use python by default, right? If so, I don't know if it would be worth adding a note here to mention that for using openPMD-api to read output data in jupyter notebooks, then one would have to install the package in the conda environments with python. Instead of just load this spack environment ... Or maybe this is very obvious! Feel free to ignore this if you think so.

Reply:
This might be a bit confusing.
This is a developer environment to build a local version of WarpX, e.g. from `git`. As such, it does not use the mainline `package.py`.

There is another workflow called `dev-build` and `build-env` in Spack that can rely on the `package.py` but we are not yet doing that one here. The reason for that is that I have to finish some usability updates in Spack before I can recommend and introduce that alternative way of building environments. (See last slide of my recent intro with outlook/alternatives.)

Reply:
Ah ok ok. Thanks for explaining it!